src/containers/Tenant/Diagnostics/Partitions/Partitions.tsx

```diff
@@ -0,0 +1,259 @@
+import block from 'bem-cn-lite';
+import {useCallback, useEffect, useMemo, useState} from 'react';
+import {useDispatch} from 'react-redux';
+import {escapeRegExp} from 'lodash/fp';
+
+import DataTable from '@gravity-ui/react-data-table';
+import {Select, TableColumnSetup} from '@gravity-ui/uikit';
+import {TableColumnSetupItem} from '@gravity-ui/uikit/build/esm/components/Table/hoc/withTableSettings/withTableSettings';
+
+import type {EPathType} from '../../../../types/api/schema';
+
+import {useAutofetcher, useTypedSelector} from '../../../../utils/hooks';
+import {DEFAULT_TABLE_SETTINGS, PARTITIONS_SELECTED_COLUMNS_KEY} from '../../../../utils/constants';
+
+import {getSettingValue, setSettingValue} from '../../../../store/reducers/settings';
+import {
+    getConsumer,
+    getPreparedPartitionsData,
+    setDataWasNotLoaded,
+} from '../../../../store/reducers/consumer';
+
+import {TableSkeleton} from '../../../../components/TableSkeleton/TableSkeleton';
+import {Search} from '../../../../components/Search';
+import {ResponseError} from '../../../../components/Errors/ResponseError';
+
+import {isCdcStreamEntityType} from '../../utils/schema';
+
+import type {IPreparedPartitionDataWithHosts} from './utils/types';
+import {
+    PARTITIONS_COLUMNS_IDS,
+    PARTITIONS_COLUMNS_TITILES,
+    PARTITIONS_DEFAULT_SELECTED_COLUMNS,
+} from './utils/constants';
+
+import {columns as partitionsColumns} from './columns';
+
+import i18n from './i18n';
+
+import './Partitions.scss';
+
+export const b = block('ydb-diagnostics-partitions');
+
+interface PartitionsProps {
+    path?: string;
+    type?: EPathType;
+    nodes?: Record<number, string>;
+    consumers?: string[];
+}
+
+export const Partitions = ({path, type, nodes, consumers}: PartitionsProps) => {
```

Comment:
this component still looks heavy
i don't think it's worth refactoring it too much right now, but if in the future it becomes more complex, maybe split it into a several components, maybe move the filtration logic into selectors (if it works)

Reply:
I did this way because I wasn’t sure how these reducers will be interconnected in the future. I supposed, there will be two components for `consumer` reducer with shared selectors and data processing logic so I kept all filters inside component. 

Further there will be a need to move some filters to the store as they will be passed to the url to enable navigating to `Consumers` tab with specific consumer selected. But I will address it in another PR